[nnc][tests] Skip long running tests when using TE interpreter

bertmaher opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#57568 [nnc][tests] Skip long running tests when using TE interpreter**

Differential Revision: [D28202740](https://our.internmc.facebook.com/intern/diff/D28202740)

bertmaher committed:
[nnc][tests] Skip long running tests when using TE interpreter

[ghstack-poisoned]

walterddr commented on test/test_jit_fuser_te.py:
```diff
@@ -1815,7 +1815,12 @@ def fn(x):
 
     def test_superslomo(self):
-        for device in self.devices:
+        devices = []
```
should we do 
```
devices = self.devices
```
if not RUN_CUDA and "cuda" in devices:
    devices.remove("cuda")
```

bertmaher replied:
Oh sure, that would work

bertmaher replied:
Ah, I think I need to do
```
devices = self.devices.copy()
```
or else this ends up mutating `self.devices`

bertmaher committed:
Update on "[nnc][tests] Skip long running tests when using TE interpreter"


Differential Revision: [D28202740](https://our.internmc.facebook.com/intern/diff/D28202740)

[ghstack-poisoned]

bertmaher committed:
Update on "[nnc][tests] Skip long running tests when using TE interpreter"


Differential Revision: [D28202740](https://our.internmc.facebook.com/intern/diff/D28202740)

[ghstack-poisoned]
